python/ray/serve/tests/unit/test_deployment_state.py

```diff
@@ -41,6 +41,8 @@
     get_random_string,
 )
 
+dead_replicas_context = set()
```

Comment:
instead of using a global set can you just make an internal call to the replica wrapper to mark that it should not recover?

Reply:
Got it, I see the issue.

In this case, I would just focus on making this set scoped to the test. So some options would be to:

1. make it an attribute of the `mock_deployment_state_manager`
2. leave it as a global but have a fixture that clears it after each test

Reply:
Got it, thanks for the suggestions! I've gone with (2) for now, and I also added a comment explaining the `dead_replicas_context` global state.